cfme/infrastructure/resource_pool.py

```diff
@@ -53,72 +136,129 @@ def _get_context(self):
             context['provider'] = self.provider
         return context
 
-    def delete(self, cancel=True):
+    def delete(self, cancel=True, wait=False):
         """Deletes a resource pool from CFME
 
-        Args:
-            cancel: Whether to cancel the deletion, defaults to True
+        :param cancel: Whether or not to cancel the deletion, defaults to True
+        :param wait: Whether or not to wait for the delete, defaults to False
         """
-        navigate_to(self, 'Details')
-        cfg_btn('Remove from the VMDB', invokes_alert=True)
-        sel.handle_alert(cancel=cancel)
+        view = navigate_to(self, 'Details')
+        view.toolbar.configuration.item_select('Remove from the VMDB', handle_alert=not cancel)
+
+        # cancel doesn't redirect, confirmation does
+        view.flush_widget_cache()
+        if cancel:
+            view = self.create_view(ResourcePoolDetailsView)
+        else:
+            view = self.create_view(ResourcePoolAllView)
+        wait_for(lambda: view.is_displayed, fail_condition=False, num_sec=10, delay=1)
+
+        # flash message only displayed if it was deleted
+        if not cancel:
+            view.entities.flash.assert_no_error()
+            view.entities.flash.assert_success_message('Delete initiated for 1 Resource Pool')
+
+        if wait:
+            self.wait_for_delete()
 
     def wait_for_delete(self):
-        navigate_to(self, 'All')
-        wait_for(lambda: not self.exists, fail_condition=False,
-                 message="Wait resource pool to disappear", num_sec=500, fail_func=sel.refresh)
+        """Wait for the delete to complete"""
+        view = navigate_to(self, 'All')
+
+        def refresh():
+            if self.provider:
+                self.provider.refresh_provider_relationships()
+            view.browser.selenium.refresh()
+            view.flush_widget_cache()
+
+        try:
+            wait_for(lambda: view.paginator.find_row_on_pages(view.entities.table,
+                                                              name=self.name).is_displayed,
+                     fail_condition=False, message='Wait resource pool to disappear', num_sec=500,
+                     fail_func=refresh)
+        except NoSuchElementException:
```

Comment:
I think its appropriate to let the TimedOutError through to the caller, since they're calling a wait_for convenience method. That or get rid of this and `wait_for_appear` convenience methods entirely since we already provide `exists`